Oss 2.4 release notes (#4352)
influxdata/docs-v2: content/influxdb/v2.4/reference/release-notes/influxdb.md

weight: 101
---

## v2.4 [2022-08-18]

### Features

- Support [InfluxDB 1.x dbrp](/influxdb/v2.4/query-data/influxql/dbrp) names as bucket IDs in `replications` API.
- Add the [InfluxQL shell (REPL)](/influxdb/v2.4/reference/cli/influx/v1/shell/).
- Change to operator model so that `admin` user has instance-level permissions without `operator` token.

#### Flux advancement highlights

> - Add multiple new join functions to the [`join package`](/flux/v0.x/stdlib/join/), including [join.full()](/flux/v0.x/stdlib/join/full/).

Reviewer: ```suggestion
- Add the [`join package`](/flux/v0.x/stdlib/join/) to support inner joins, right outer joins, left outer joins, and full outer joins.
```

- Promote experimental.to() to [influxdb.wideTo()](/flux/v0.x/stdlib/influxdata/influxdb/wideto/).

> - Add [`initialZero`](/flux/v0.x/stdlib/universe/derivative/#initialzero) parameter to [derivative() function](/flux/v0.x/stdlib/universe/derivative/).

Reviewer: ```suggestion
- Add [`initialZero`](/flux/v0.x/stdlib/universe/derivative/#initialzero) parameter to [`derivative()`](/flux/v0.x/stdlib/universe/derivative/).
```

> - Add builtin function `time` to the [`date package`](flux/v0.x/stdlib/date/) to convert `timeable` into `datetime`.

Reviewer: ```suggestion
- Add `time()` function to the [`date package`](flux/v0.x/stdlib/date/) to convert timeable types (time and duration) to time types.
```

> - Add builtin function `time` to the [`date package`](flux/v0.x/stdlib/date/) to convert `timeable` into `datetime`.

Author: @sanderson is it timeable or time.able?

> - Add builtin function `time` to the [`date package`](flux/v0.x/stdlib/date/) to convert `timeable` into `datetime`.

Reviewer: It's "timeable." It describes a type constraint in flux: https://docs.influxdata.com/flux/v0.x/spec/types/#timeable-constraint

These a types that can be cast into a time value.

- Promote the following functions from experimental.array into the [`array package`](/flux/v0.x/stdlib/array/): 
  - [array.concat()](/flux/v0.x/stdlib/array/concat/)
  - [array.filter()](/flux/v0.x/stdlib/array/filter/)
  - [array.map()](/flux/v0.x/stdlib/array/map/)
- Promote the following functions from experimental.http.requests into the [`http.requests package`](/flux/v0.x/stdlib/http/requests/): 

>   - [http.requests.do()](/flux/v0.x/stdlib/http/requests/do/) 
>   - [http.requests.get()](/flux/v0.x/stdlib/http/requests/get/)
>   - [http.requests.peek()](/flux/v0.x/stdlib/http/requests/peek/)
>   - [http.requests.post()](/flux/v0.x/stdlib/http/requests/post/)

Reviewer: ```suggestion
  - [requests.do()](/flux/v0.x/stdlib/http/requests/do/) 
  - [requests.get()](/flux/v0.x/stdlib/http/requests/get/)
  - [requests.peek()](/flux/v0.x/stdlib/http/requests/peek/)
  - [requests.post()](/flux/v0.x/stdlib/http/requests/post/)
```

> - Promote experimental.bitwise into the [`bitwise package`](/flux/v0.x/stdlib/bitwise/). 

Reviewer: ```suggestion
- Promote `experimental.bitwise()` into the [`bitwise` package](/flux/v0.x/stdlib/bitwise/). 
```

- Add new [experimental.catch()](/flux/v0.x/stdlib/experimental/catch/) and [testing.shouldError()](/flux/v0.x/stdlib/testing/shoulderror/) functions.

> - Support conditional expressions, constants, and literals in vectorized [map()](flux/v0.x/stdlib/universe/map/).

Reviewer: ```suggestion
- Support conditional expressions, constants, and literals in vectorized [`map()`](flux/v0.x/stdlib/universe/map/).
```

> - Optimize [holtWinters() function](/flux/v0.x/stdlib/universe/holtwinters/) and redundant sorts.

Reviewer: ```suggestion
- Optimize [`holtWinters()`](/flux/v0.x/stdlib/universe/holtwinters/) and redundant sorts.
```

> - Deprecate experimental.http.get() and experimental.csv.from().

Reviewer 2: ```suggestion
- Deprecate `experimental.http.get()` and `experimental.csv.from()`.
```

### Bug fixes

- Log the log level at startup.
- Fix panic from `CreateIterator` in Flux queries.
- Fix error-caching in `bufio.Writer`.
- Remove `MATCHER_BEHAVIOR` environment variable.

### Maintenance 

- Upgrade to [Go 1.18.4](https://go.dev/doc/go1.18).

> - Upgrade to [Flux 0.177.0](/flux/v0.x/release-notes/#v01770-2022-08-01).

Reviewer 3: ```suggestion
- Upgrade to [Flux 0.179.0](flux/v0.x/release-notes/#v01790-2022-08-15).
```

weight: 101
---

## v2.4 [2022-08-18]

### Features

> - Support [InfluxDB 1.x dbrp](/influxdb/v2.4/query-data/influxql/dbrp) names as bucket IDs in `replications` API.

Reviewer: ```suggestion
- Support bucket names in the `replications` API and allow [InfluxDB 1.x DBRP](/influxdb/v2.4/query-data/influxql/dbrp) names as bucket names.
```

- Add the [InfluxQL shell (REPL)](/influxdb/v2.4/reference/cli/influx/v1/shell/).
- Change to operator model so that `admin` user has instance-level permissions without `operator` token.

#### Flux advancement highlights

> - Add the [`join package`](/flux/v0.x/stdlib/join/) to support inner joins, right outer joins, left outer joins, and full outer joins.

Reviewer: ```suggestion
- Add the [`join` package](/flux/v0.x/stdlib/join/) to support inner joins, right outer joins, left outer joins, and full outer joins.
```

- Promote `experimental.to()` to [`influxdb.wideTo()`](/flux/v0.x/stdlib/influxdata/influxdb/wideto/).
- Add [`initialZero`](/flux/v0.x/stdlib/universe/derivative/#initialzero) parameter to [`derivative()`](/flux/v0.x/stdlib/universe/derivative/).
- Add `time()` function to the [`date package`](flux/v0.x/stdlib/date/) to convert timeable types (time and duration) to time types.
- Promote the following functions from `experimental.array` into the [`array` package](/flux/v0.x/stdlib/array/): 

>   - [`array.concat(`)](/flux/v0.x/stdlib/array/concat/)

Reviewer: ```suggestion
  - [`array.concat()`](/flux/v0.x/stdlib/array/concat/)
```

>   - [array.concat()](/flux/v0.x/stdlib/array/concat/)
>   - [array.filter()](/flux/v0.x/stdlib/array/filter/)
>   - [array.map()](/flux/v0.x/stdlib/array/map/)

Reviewer: ```suggestion
  - [`array.concat(`)](/flux/v0.x/stdlib/array/concat/)
  - [`array.filter()`](/flux/v0.x/stdlib/array/filter/)
  - [`array.map()`](/flux/v0.x/stdlib/array/map/)
```

- Promote the following functions from the experimental `http.requests` package into the [`http/requests` package](/flux/v0.x/stdlib/http/requests/): 

>   - [requests.do()](/flux/v0.x/stdlib/http/requests/do/) 
>   - [requests.get()](/flux/v0.x/stdlib/http/requests/get/)
>   - [requests.peek()](/flux/v0.x/stdlib/http/requests/peek/)
>   - [requests.post()](/flux/v0.x/stdlib/http/requests/post/)

Reviewer: ```suggestion
  - [`requests.do()`](/flux/v0.x/stdlib/http/requests/do/) 
  - [`requests.get()`](/flux/v0.x/stdlib/http/requests/get/)
  - [`requests.peek()`](/flux/v0.x/stdlib/http/requests/peek/)
  - [`requests.post()`](/flux/v0.x/stdlib/http/requests/post/)
```

- Promote `experimental.bitwise()` into the [`bitwise` package](/flux/v0.x/stdlib/bitwise/). 
- Add new [experimental.catch()](/flux/v0.x/stdlib/experimental/catch/) and [testing.shouldError()](/flux/v0.x/stdlib/testing/shoulderror/) functions.
- Support conditional expressions, constants, and literals in vectorized [`map()`](flux/v0.x/stdlib/universe/map/).
- Optimize [`holtWinters()`](/flux/v0.x/stdlib/universe/holtwinters/) and redundant sorts.
- Deprecate the following experimental functions:
  - `http.get()`

>   - `csv.from()`.

Reviewer: ```suggestion
  - `csv.from()`
```

### Bug fixes

- Log the log level at startup.
- Fix panic from `CreateIterator` in Flux queries.

> - Fix error-caching in `bufio.Writer`.

Reviewer 4: this seems to have ended up here by mistake, no such change was merged for what i can see.
the original [draft](https://github.com/influxdata/docs-v2/issues/4255) references my MR https://github.com/influxdata/influxdb/pull/23127 which has not been merged.
the entry was re-worded to "Fix error-caching in `bufio.Writer`",
but bufio.Writer is part of the go standard library (in which has no related change is to be found.)
